core/src/main/java/org/apache/calcite/sql/type/OperandTypes.java

```diff
@@ -510,6 +510,38 @@ public static SqlOperandTypeChecker variadic(
         }
       };
 
+  public static final SqlSingleOperandTypeChecker ARRAY_OF_INTEGER =
```

Comment:
Just to clarify, what is the benefit of adding this if you just end up checking that each array element is a number in the `SqlFunctions` method you added?

Reply:
Sounds good, thanks for explaining!